### ISSUE TITLE
Update __repr__ for transforms

### ISSUE THREAD

**Issue by vishwakftw** [pytorch/vision#418]
PR https://github.com/pytorch/vision/pull/380 adds `__repr__` for existing transforms. However, optional parameters have been missed out in the process.

By resolving this issue, I would like to add those optional parameters in the `__repr__` definitions.

For `Normalize`, I noticed that we have to give a sequence. In case we gave a `Number` for either `mean` or `std`, we could effectively elongate it (broadcast), and use it. Currently, if users wanted to normalize with the same value in all the channels, then they would have to create a list containing same entries. I am thinking about a behaviour similar to https://github.com/pytorch/vision/blob/master/torchvision/transforms/transforms.py#L189-L191. 

Please let me know what you think.

cc: @fmassa @alykhantejani

**Comment by fmassa** [comment]
Yes, let's fix the `__repr__`, thanks for catching that!

About extending the behaviour of `Normalize`, I currently like the way it's implemented because it is very explicit what the parameters are. I'm not sure that the case where the same value is present for all channels is that common (but I might be wrong), and having it explicit might save people from bugs (say they wanted to have channel-wise mean, but by mistake did `torch.mean(imgs)`, which reduces to a single number).

@alykhantejani what do you think?

**Comment by alykhantejani** [comment]
@vishwakftw nice catch with the `__repr__`! LGTM

I agree with @fmassa about the explicitness of `Normalize`, sometimes auto-broadcasting can create subtle bugs in your code that can take hours to find.

**Comment by vishwakftw** [comment]
Thanks @fmassa and @alykhantejani . I think I will work on updating `__repr__` for transforms then.
